panther_power_control/src/relays_node.py

```diff
@@ -134,19 +149,50 @@ def _e_stop_reset_cb(self, req: TriggerRequest) -> TriggerResponse:
             return TriggerResponse(True, 'E-STOP reset successful')
 
     def _e_stop_trigger_cb(self, req: TriggerRequest) -> TriggerResponse:
-        with self._lock:
+        with self._e_stop_lock:
             if self._e_stop_state:
                 return TriggerResponse(True, 'E-SROP already triggered')
-            
+
             self._e_stop_state = True
             self._e_stop_state_pub.publish(self._e_stop_state)
             return TriggerResponse(True, 'E-SROP triggered successful')
 
+    def _motor_enable_cb(self, req: SetBoolRequest) -> SetBoolResponse:
+        with self._motors_lock:
+            if not self._lines['STAGE2_INPUT'].get_value():
+                self._motor_enabled = False
+                return SetBoolResponse(
+                    not req.data,
+                    f'Motors are {"already " if not req.data else ""}disabled. '
+                    + '(Main switch set to Stage 1)',
+                )
+
+            if self._motor_enabled == req.data:
+                return SetBoolResponse(
+                    True, f'Motors are already {"enabled" if self._motor_enabled else "disabled"}'
+                )
+
+            self._motor_enabled = req.data
+            self._lines['MOTOR_ON'].set_value(req.data)
+            self._publish_motor_state(req.data)
+            return SetBoolResponse(True, f'Motors {"enabled" if req.data else "disabled"}')
+
     def _set_motor_state_timer_cb(self, *args) -> None:
-        motor_state = self._lines['STAGE2_INPUT'].get_value()
-        self._lines['MOTOR_ON'].set_value(motor_state)
-        if self._io_state.motor_on != motor_state:
-            self._io_state.motor_on = motor_state
+        with self._motors_lock:
+            motor_state = self._lines['STAGE2_INPUT'].get_value()
+            # if switch changes from off to on overwrite service value
+            if not self._last_motor_state and motor_state:
+                self._motor_enabled = True
+
+            self._last_motor_state = motor_state
```

Comment:
Taught with C++ practice, I think that it would not be a stupid approach to use static variables in such situations. The `_last_motor_state` variable is really only needed in this function, so you can encapsulate such things so as not to clutter the whole class.

Example
```python
def my_function(self):
    if not hasattr(self.my_function, "counter"):
        self.my_function.counter = 0

    self.my_function.counter += 1
    print(f"Counter: {self.my_function.counter}")
```

Reply:
I don't see the point. This makes the code less readable. We would have to make an strong assumption the class is a singleton. Even though it would work I do not think assigning variables to a function rather than object is a good idea